### PR TITLE
add support for nvidia/nv-embed-v1 (https://huggingface.co/nvidia/NV-Embed-v1)

### PR DESCRIPTION
**Pull request by mattf**
# Description

Add support for https://huggingface.co/nvidia/NV-Embed-v1

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [ ] Yes
- [x] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [x] Yes
- [ ] No

## Type of Change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense
- [x] Covered by existing unit/integration tests

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-integrations/embeddings/llama-index-embeddings-nvidia/llama_index/embeddings/nvidia/base.py [hunk 1]
```diff
@@ -14,12 +14,17 @@
...
+# integrate.api.nvidia.com is the default url for most models, any
+# bespoke endpoints will need to be added to the MODEL_ENDPOINT_MAP
+BASE_URL = "https://integrate.api.nvidia.com/v1/"
 DEFAULT_MODEL = "NV-Embed-QA"
 
+# because MODEL_ENDPOINT_MAP is used to construct KNOWN_URLS, we need to
+# include at least one model w/ https://integrate.api.nvidia.com/v1/
 MODEL_ENDPOINT_MAP = {
-    DEFAULT_MODEL: BASE_URL,
+    "NV-Embed-QA": "https://ai.api.nvidia.com/v1/retrieval/nvidia/",
```
Comment (ravi03071991) [inline llama-index-integrations/embeddings/llama-index-embeddings-nvidia/llama_index/embeddings/nvidia/base.py:25]: is NV-Embed-QA  an embedding model?

Comment (mattf) [reply llama-index-integrations/embeddings/llama-index-embeddings-nvidia/llama_index/embeddings/nvidia/base.py:25]: it is, see https://build.nvidia.com/nvidia/embed-qa-4?snippet_tab=Shell

Comment (ravi03071991) [reply llama-index-integrations/embeddings/llama-index-embeddings-nvidia/llama_index/embeddings/nvidia/base.py:25]: Cool. Thanks.

**Review by ravi03071991 — APPROVED** [review]
LGTM.

**Pull request merged**